### PR TITLE
migrate from intltool to gettext

### PR DESCRIPTION
**Pull request by yetist**
Please review and test.

### CODE REVIEW

**Comment by raveit65** [comment]
We have a file conflict.

**Comment by raveit65** [comment]
@yetist 
file conflict.

**Comment by raveit65** [comment]
autogen.sh was added to EXTRA_DIST https://github.com/mate-desktop/mate-user-share/commit/28b2950f617049b2ed83a40721299d415bb8a637

File: data/libcaja-user-share.caja-extension.in.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [Caja Extension]
 Icon=user-share
-_Name=User Share
-_Description=Integrates with user-share
+Name=User Share
+Description=Integrates with user-share
 Copyright=Copyright (C) 2005 William Jon McCann <mccann@jhu.edu>
```
Comment (raveit65) [inline data/libcaja-user-share.caja-extension.in.in]: Looks like that this translations are missing.
```
diff -uprN mate-user-share-orig.pot mate-user-share.pot > diff-pot-po-dir.diff

<cut>
-#: ../data/libcaja-user-share.caja-extension.in.in.h:1
-msgid "User Share"
+#: data/mate-user-share-properties.desktop.in:3 src/caja-share-bar.c:99
+msgid "Personal File Sharing"
 msgstr ""
 
-#: ../data/libcaja-user-share.caja-extension.in.in.h:2
-msgid "Integrates with user-share"
+#: data/mate-user-share-properties.desktop.in:4
+msgid "Preferences for sharing of files"
 msgstr ""

<cut>
```
All other translations are there.

**Comment by raveit65** [comment]
@yetist 
Does this work for you?
Here the translation is missing, same like before.
diff between master and getext branch.
```
./autogen.sh && ./makepot

diff -uprN mate-user-share.pot.master mate-user-share.pot > mate-user-share.pot.diff
<cut>
-#: ../data/libcaja-user-share.caja-extension.in.in.h:1
-msgid "User Share"
+#: data/mate-user-share-properties.desktop.in:3 src/caja-share-bar.c:99
+msgid "Personal File Sharing"
 msgstr ""
 
-#: ../data/libcaja-user-share.caja-extension.in.in.h:2
-msgid "Integrates with user-share"
+#: data/mate-user-share-properties.desktop.in:4
+msgid "Preferences for sharing of files"
 msgstr ""

<cut>

**Comment by yetist** [comment]
Try to update pot and po files.

**Comment by sc0w** [comment]
new warnings detected:

```
/usr/bin/xgettext: warning: The option --msgid-bugs-address was not specified.
                            If you are using a 'Makevars' file, please specify
                            the MSGID_BUGS_ADDRESS variable there; otherwise please
                            specify an --msgid-bugs-address command line option.

/usr/bin/xgettext: warning: file 'data/libcaja-user-share.caja-extension.in.in' extension 'caja-extension' is unknown; will try C
```

**Comment by raveit65** [comment]
Resoure and translation for transifex are updated with minimal changes, PR is rebased.

**Comment by raveit65** [comment]
No, updating pot + po files doesn't help.
1. on `master` branch `./autogen.sh && ./makepot`
2. `cp mate-user-share.pot mate-user-share.pot.master`
3. switch to `getext` branch
4. on `getext` branch `./autogen.sh && ./makepot`
5. `diff -uprN mate-user-share.pot.master mate-user-share.pot > mate-user-share.pot.diff`
Result is:
```
-#: ../data/libcaja-user-share.caja-extension.in.in.h:1
-msgid "User Share"
+#: data/mate-user-share-properties.desktop.in:3 src/caja-share-bar.c:99
+msgid "Personal File Sharing"
 msgstr ""
 
-#: ../data/libcaja-user-share.caja-extension.in.in.h:2
-msgid "Integrates with user-share"
+#: data/mate-user-share-properties.desktop.in:4
+msgid "Preferences for sharing of files"
 msgstr ""
```

**Comment by yetist** [comment]
I think the msgfmt and xgettext are ok now, please test again.

**Comment by raveit65** [comment]
> I think the msgfmt and xgettext are ok now, please test again.

I did test it already with your change.
```
[rave@mother mate-user-share-dist]$ git log -p
commit 1e2d195b83eb51f44352876dc939a32ed1ece93d (HEAD -> gettext, origin/gettext)
Author: Wu Xiaotian <yetist@gmail.com>
Date:   Mon Jul 22 11:16:12 2019 +0800

    Let Description in .caja-extension be translatable

diff --git a/data/Makefile.am b/data/Makefile.am
index eda2d4f..61d85c5 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -21,7 +21,7 @@ extensiondir = $(datadir)/caja/extensions
 extension_in_files = libcaja-user-share.caja-extension.in
 extension_DATA = $(extension_in_files:.caja-extension.in=.caja-extension)
 %.caja-extension: %.caja-extension.in
-       $(AM_V_GEN) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+       $(AM_V_GEN) $(MSGFMT) --desktop --keyword=Name --keyword=Description --template $< -d $(top_srcdir)/po -o $@
 
 EXTRA_DIST = $(gsettings_SCHEMAS) \
        $(app_DATA)             \
diff --git a/po/Makevars b/po/Makevars
index 5c3836c..512e929 100644
--- a/po/Makevars
+++ b/po/Makevars
@@ -8,7 +8,7 @@ subdir = po
 top_builddir = ..
 
 # These options get passed to xgettext.
-XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --add-comments
+XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --add-comments --keyword=Description
 
 # This is the copyright holder that gets inserted into the header of the
 # $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
```

**Comment by yetist** [comment]
I know the reason: `xgettext` does not think `caja-extension.in` is a desktop file, and ignores it when making a pot file.

Now only msgfmt can generate multilingual entries for Description.

**Comment by raveit65** [comment]
Cool , this trick fixes the problem.
```
-#: ../data/libcaja-user-share.caja-extension.in.in.h:1
+#: data/libcaja-user-share.desktop.in.in:3
+msgid "user-share"
+msgstr ""
+
+#: data/libcaja-user-share.desktop.in.in:4
 msgid "User Share"
 msgstr ""
 
-#: ../data/libcaja-user-share.caja-extension.in.in.h:2
+#: data/libcaja-user-share.desktop.in.in:5
 msgid "Integrates with user-share"
 msgstr ""
```
Installation of this file is same like before.
```
/usr/bin/install -c -m 644 libcaja-user-share.caja-extension '/usr/local/share/caja/extensions'
```
I am still wondering that `Icon=user-share` will be translated too, but this happens with other desktop files too.
Eg. `Icon=folder-remote`
```
-#: ../data/mate-user-share-obexftp.desktop.in.in.h:1
+#: data/mate-user-share-properties.desktop.in:6
+#: data/mate-user-share-obexftp.desktop.in.in:7
+#: data/mate-user-share-obexpush.desktop.in.in:7
+#: data/mate-user-share-webdav.desktop.in.in:7
+msgid "folder-remote"
+msgstr ""
```
Btw. what happens when `Icon=user-share` or `Icon=folder-remote` will be translated. Will the desktop file file find the icon?

We need to check this in other package where port to gettext is done.

**Comment by yetist** [comment]
The new version of xgettext allows Icon to be translatable.
```bash
$ cat -n a.desktop 
     1	[TITLE]
     2	Name=Atril Document Viewer
     3	GenericName=Document Viewer
     4	Comment=View multi-page documents
     5	TryExec=apps
     6	Exec=apps
     7	Description=The Description
     8	Icon=atril
$ xgettext --keyword=Description a.desktop -o -
# SOME DESCRIPTIVE TITLE.
# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
# This file is distributed under the same license as the PACKAGE package.
# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
#
#, fuzzy
msgid ""
msgstr ""
"Project-Id-Version: PACKAGE VERSION\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2019-07-23 23:02+0800\n"
"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
"Language-Team: LANGUAGE <LL@li.org>\n"
"Language: \n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=CHARSET\n"
"Content-Transfer-Encoding: 8bit\n"

#: a.desktop:3
msgid "Atril Document Viewer"
msgstr ""

#: a.desktop:4
msgid "Document Viewer"
msgstr ""

#: a.desktop:5
msgid "View multi-page documents"
msgstr ""

#: a.desktop:8
msgid "The Description"
msgstr ""

#: a.desktop:9
msgid "atril"
msgstr ""

```

**Comment by raveit65** [comment]
@yetist 
We forget to add translator hints to desktop files during review.
```
#: data/mate-user-share-properties.desktop.in:6
#: data/mate-user-share-obexftp.desktop.in.in:7
#: data/mate-user-share-obexpush.desktop.in.in:7
#: data/mate-user-share-webdav.desktop.in.in:7
msgid "folder-remote"
msgstr ""

#: data/mate-user-share-properties.desktop.in:11
msgid ""
"MATE;personal;file;sharing;preferences;properties;settings;WebDAV;ObexFTP;"
msgstr ""
```

**Comment by raveit65** [comment]
Done!
https://github.com/mate-desktop/mate-user-share/commit/5f1ab20776fb303d6ad237b290507dc927efe884

**Pull request merged**